packages/loader/container-loader/src/container.ts

```diff
@@ -1964,6 +1964,7 @@ export class Container
 				true, // batch
 				message.metadata,
 				message.compression,
+				referenceSequenceNumber ?? message.referenceSequenceNumber,
```

Comment:
This feels questionable to me. The way interface is structured right now - it basically says we are Ok allowing different ref sequence numbers on ops in a batch, but we will also provide an overwrite for whole batch. Runtime, if it wants to, can overwrite ref seq number on every op.
I'd not add IBatchMessage.referenceSequenceNumber, and instead this code use either ref seq number provided by runtime (for whole batch) or default one from delta manager. And in future (next) releases remove capability for later (i.e. make argument required).

Reply:
Thanks for calling this out, I had my doubts about creating too many states.